controllers/controllers/workloads/cforg_controller.go

```diff
@@ -0,0 +1,172 @@
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the CFOrg object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
+func (r *CFOrgReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	cfOrg := new(workloadsv1alpha1.CFOrg)
+	err := r.Client.Get(ctx, req.NamespacedName, cfOrg)
+	if err != nil {
+		r.Log.Error(err, fmt.Sprintf("Error when trying to fetch CFOrg %s/%s", req.Namespace, req.Name))
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if cfOrg.ObjectMeta.DeletionTimestamp != nil && !cfOrg.ObjectMeta.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, nil
+	}
+
+	var anchor v1alpha2.SubnamespaceAnchor
+	err = r.Client.Get(ctx, req.NamespacedName, &anchor)
+	if err != nil {
+		if !k8serrors.IsNotFound(err) {
+			r.Log.Error(err, fmt.Sprintf("Error getting SubnamespaceAnchor for CFOrg %s/%s", req.Namespace, req.Name))
+			return ctrl.Result{}, err
+		}
+
+		anchor = v1alpha2.SubnamespaceAnchor{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      req.Name,
+				Namespace: req.Namespace,
+				Labels: map[string]string{
+					OrgNameLabel: cfOrg.Spec.Name,
+				},
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: APIVersion,
+						Kind:       Kind,
+						Name:       cfOrg.Name,
+						UID:        cfOrg.GetUID(),
+					},
+				},
+			},
+		}
+
+		err = r.Client.Create(ctx, &anchor)
+		if err != nil {
+			r.Log.Error(err, fmt.Sprintf("Error creating SubnamespaceAnchor for CFOrg %s/%s", req.Namespace, req.Name))
+			return ctrl.Result{}, err
+		}
+
+		setStatusConditionOnLocalCopy(&cfOrg.Status.Conditions, StatusConditionReady, metav1.ConditionUnknown, StatusConditionReady, "")
+		err = r.Client.Status().Update(ctx, cfOrg)
+		if err != nil {
+			r.Log.Error(err, "unable to update CFOrg status")
+			r.Log.Info(fmt.Sprintf("CFOrg status: %+v", cfOrg.Status))
+			return ctrl.Result{}, err
+		}
+	}
+
+	if anchor.Status.State != v1alpha2.Ok {
+		return ctrl.Result{RequeueAfter: 100 * time.Millisecond}, nil
+	}
+
+	namespace := new(corev1.Namespace)
+	err = r.Client.Get(ctx, types.NamespacedName{Name: cfOrg.ObjectMeta.Name}, namespace)
+	if err != nil {
+		return ctrl.Result{RequeueAfter: 100 * time.Millisecond}, nil
+	}
+
+	err = setCascadingDelete(ctx, r.Client, req.Name)
+	if err != nil {
+		r.Log.Error(err, fmt.Sprintf("Error updating HierarchyConfiguration for CFOrg %s/%s", req.Namespace, req.Name))
+		return ctrl.Result{}, err
+	}
+
+	cfOrg.Status.GUID = namespace.Name
+	setStatusConditionOnLocalCopy(&cfOrg.Status.Conditions, StatusConditionReady, metav1.ConditionTrue, StatusConditionReady, "")
```

Comment:
We tried to do this, but to make it usable for all types, we'd have to use a generic client.Object and that won't let us access the status. In addition, we also have to pass in the logger and client from the reconciler. The setStatusCOnditionOnLocalCopy is a repeat of the pattern in the other controllers since that seemed like the best compromise in making a shared function that didn't require passing in all the objects from the reconcilers.